pandas/core/arrays/period.py

```diff
@@ -1,35 +1,54 @@
 # -*- coding: utf-8 -*-
 from datetime import timedelta
 import operator
-import warnings
 
 import numpy as np
 
+from pandas import compat
+from pandas.compat.numpy import function as nv
 from pandas._libs import lib
 from pandas._libs.tslib import NaT, iNaT
 from pandas._libs.tslibs.period import (
     Period, IncompatibleFrequency, DIFFERENT_FREQ_INDEX,
-    get_period_field_arr, period_asfreq_arr)
+    get_period_field_arr, period_asfreq_arr,
+)
 from pandas._libs.tslibs import period as libperiod
 from pandas._libs.tslibs.timedeltas import delta_to_nanoseconds, Timedelta
 from pandas._libs.tslibs.fields import isleapyear_arr
+from pandas.util._decorators import cache_readonly
```

Comment:
can you isort (if you have not done), and remov from the non-checking list

Reply:
Going to wait on that since there are other outstanding PRs touching imports.